docs/src/2.4.ManagingTheCamera.mdx

```diff
@@ -0,0 +1,173 @@
+  );
+}
+```
+
+To move the camera, we will need to create a state that gets updated over time. [window.requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) generally matches the display refresh rate. Let's add a [React Hook](https://reactjs.org/docs/hooks-intro.html) that accepts a callback which will be called before each browser repaint.
+
+```js
+// src/utils/useRequestAnimationFrame.js
+import { useLayoutEffect, useRef, useEffect, useCallback } from "react";
+
+// a react hook that accepts a callback function which will be called before each browser repaint
+export default function useRequestAnimationFrame(callback) {
+  // use ref so the callback will persist for the full lifetime of the component
+  const callbackRef = useRef(callback);
+  // only update callbackRef when the callback changes
+  useEffect(() => (callbackRef.current = callback), [callback]);
+
+  // persist the requestAnimationFrameId so the last refresh callback can be cancelled when component unmounts
+  const rafId = useRef();
+  // the function to be called in every requrestAnimationFrame
+  function loop() {
+    rafId.current = requestAnimationFrame(loop);
+    callbackRef.current();
+  }
+
+  // start the loop when the component mounts and cancel the animation frame when unmounts
+  useEffect(() => {
+    rafId.current = requestAnimationFrame(loop);
+    return () => cancelAnimationFrame(rafId.current);
+  }, []);
+}
+```
+
+Lastly, let's hook everything together. We'll use `useRequestAnimationFrame` to increase a counter by `1` in every requestAnimaitonFrame callback, and use the counter to adjust the camera's `phi` and `thetaOffset` accordingly.
+
+```js
+// src/Example.js
+
+import Duck from "./Duck";
+import useRequestAnimationFrame from "./utils/useRequestAnimationFrame";
+
+export default function Example() {
+  const cameraMoveSpeed = 1 / 100;
+  // use count to derive camera phi and thetaOffset values
+  const [count, setCount] = useState(0);
+  useRequestAnimationFrame(() => {
+    // update count before each browser repaint
+    const newCount = count + 2 >= steps ? 0 : count + 1;
+    setCount(newCount);
+  });
+
+  // generate cameraState from `count`
+  const cameraState = {
+    ...DEFAULT_CAMERA_STATE,
+    phi: count * cameraMoveSpeed * Math.PI,
+    thetaOffset: count * cameraMoveSpeed * 2 * Math.PI,
+  };
+
+  // ...other existing code
+
+  return (
+    <Worldview cameraState={cameraState}>
+      <Spheres>{[marker]}</Spheres>
+      <Axes />
+      <Duck />
+    </Worldview>
+  );
+}
+```
+
+Now the camera is moving around the scene, giving us 360° view of the duck model!
+
+Notice we've added `cameraState={cameraState}`, which turns `Worldview` into a [controlled](https://reactjs.org/docs/forms.html) component. Controlled components usually require an `onChange` handler, but no `onCameraStateChange` prop is needed for our example, since we don't need to know the new internal `cameraState` value.
```

Comment:
```suggestion
Notice that we've added `cameraState={cameraState}`, which turns `Worldview` into a [controlled component](https://reactjs.org/docs/forms.html). Controlled components usually require an `onChange` handler, but no `onCameraStateChange` prop is needed for our example, since we don't need to do anything with the new `cameraState` value.
```

Reply:
include some mention of the fact that the user can change cameraState with the mouse?